src/main.rs

```diff
@@ -164,12 +178,29 @@ async fn main() -> Result<(), errors::Error> {
 
     let args = Args::parse();
 
+    let credentials_mode = if let Some(aws_role_arn) = &args.aws_role_arn {
+        CredentialsMode::RoleBased {
+            aws_role_arn: aws_role_arn.clone(),
+        }
+    } else if let (Some(aws_access_key_id), Some(aws_secret_access_key)) =
+        (&args.aws_access_key_id, &args.aws_secret_access_key)
+    {
+        CredentialsMode::AccessKeyBased {
+            aws_access_key_id: aws_access_key_id.clone(),
+            aws_secret_access_key: aws_secret_access_key.clone(),
+        }
+    } else {
+        unreachable!()
```

Comment:
_question:_ I used unreachable here instead of the error, since it cannot happen. Please give me your opinion

Reply:
I think it's ok, but maybe the panic is better suited for such case: https://stackoverflow.com/a/11090299

Reply:
Agreed, makes it easier to debug